### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by stdavis**
@steveoh: What do you think?

### CODE REVIEW

File: PrintProxy.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+def make_request(url, webmap, print_format, layout_template):
+    response_json = requests.post(format_url(url), data={
+        'Web_Map_as_JSON': webmap,
+        'Format': print_format,
+        'Layout_Template': layout_template,
+        'f': 'json'
+    }).json()
+    if 'error' in response_json.keys():
+        raise Exception(response_json['error'])
```
Comment (steveoh) [inline PrintProxy.py:36]: what happens here if we raise an exception? Will the calling code get the information it needs?

File: PrintProxy.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    else:
+        AddMessage(response_json['results'][0]['value']['url'])
+        return response_json['results'][0]['value']['url']
+
+
+def format_url(url):
+    if not url.startswith('http'):
```
Comment (steveoh) [inline PrintProxy.py:43]: will https ever be necessary here?

Comment (stdavis) [inline PrintProxy.py:43]: Not that I can think of at the moment but we could certainly add that in the future.

File: PrintProxy.py (hunk 1 above)
Comment (stdavis) [inline PrintProxy.py:36]: I think that this depends on the messaging level of the gp service. But yes, the service would return the same error that the print service did.

**Pull request merged**